control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -1810,6 +1815,13 @@ func (r *HostedControlPlaneReconciler) generateControlPlaneManifests(ctx context
 	params.ImageRegistryHTTPSecret = generateImageRegistrySecret()
 	params.APIAvailabilityPolicy = render.SingleReplica
 	params.ControllerAvailabilityPolicy = render.SingleReplica
+	if hcp.Spec.ControllerAvailabilityPolicy == hyperv1.HighlyAvailable {
```

Comment:
was there any template actually using this? Now this is propagated to the guest cluster infra CR but Is there any component in the guest cluster that will actually use this now? If not we should probably add a comment here to clarify that.
cc @csrwng

Reply:
it was not used anywhere, same as `APIAvailabilityPolicy`, and I didn't find any place using that variable. Is it some dead code? 

I added `ControllerAvailabilityPolicy` into the `infrastructure.status.controlPlaneTopology`, although it won't affect anything 😂

Reply:
So can we completely remove `params.ControllerAvailabilityPolicy`? cc @csrwng

Reply:
Agree, I don't think it's used in the remaining templates

Reply:
cool `params.ControllerAvailabilityPolicy` is removed, and it works 😂